src/components/VideoPlayer/VideoPlayer.component.jsx

```diff
@@ -11,8 +12,33 @@ import {
   StyledVideoLike,
 } from './styled';
 
-function VideoPlayer({ videoId, title, description }) {
+function VideoPlayer({ videoId, title, description, thumbnail }) {
   const videRole = 'youtube-video';
+  const { addToFavorites, favorites, deleteFavorites } = useAuth();
+  const checkAlreadyFavorite = (currentId) => {
+    let found = false;
+    favorites.map((item) => {
+      if (item.id === currentId) {
+        found = true;
+      }
+      return item;
+    });
+    return found;
+  };
+  const handleDeleteFavorite = (event, currentId) => {
+    event.preventDefault();
+    deleteFavorites(currentId);
+  };
+  const handleAddToFavorites = (event, id, videoTitle, videoDescription) => {
+    event.preventDefault();
+    const newFavorite = {
+      id,
+      videoTitle,
+      videoDescription,
+      thumbnail,
+    };
+    addToFavorites(newFavorite);
```

Comment:
I believe this is allowing the user to add favorites without the need of being logged in.